### PR TITLE
Feature/sc 97720/salesforce event start end date error handling

### PR DESCRIPTION
**Pull request by amDavid01**
https://app.shortcut.com/deskpro/story/97720/salesforce-event-start-end-date-error-handling

### CODE REVIEW

File: src/pages/create/Activity.tsx [hunk 1]
```diff
@@ -138,11 +170,15 @@ export const CreateActivity = () => {
...
+      .then(() => navigate(-1))
+      .catch((e: Error) => {
+        setSubmissionError(mapErrorMessage(e));
 
-    navigate(-1);
+        setIsSubmitting(false);
+      });
   };
-
+  console.log(errors);
```
Comment (AshleyDawson) [inline src/pages/create/Activity.tsx]: think this is debug? please remove

File: src/utils.ts [hunk 2]
```diff
@@ -7,125 +16,157 @@ import defaultLeadLayout from "./resources/default_layout/lead.json";
...
+  const parsedErr = JSON.parse(error.message) as {
+    status: number;
+    message: string;
+  };
+  const parsedSalesforceErr = JSON.parse(parsedErr.message) as {
+    errorCode: string;
+    fields: string[];
+    message: string;
+  }[];
+  console.log(parsedSalesforceErr);
```
Comment (AshleyDawson) [inline src/utils.ts:150]: debug? please remove

File: src/utils.ts [hunk 3]
```diff
@@ -7,125 +16,157 @@ import defaultLeadLayout from "./resources/default_layout/lead.json";
...
-            return layout.list;
-        })
-        .with(["Event", "view"], () => {
-            const layout: EventLayout = settings?.mapping_note
-                ? JSON.parse(settings.mapping_note)
-                : defaultEventLayout
-            ;
-    
-            return layout.list;
+            .with(["REQUIRED_FIELD_MISSING", ["DurationInMinutes"]], () => {
```
Comment (AshleyDawson) [inline src/utils.ts:163]: Hard coded field, `DurationInMinutes`? what happens if someone changes are removes this field in Salesforce? Is that even possible? If it's not possible to remove/change this field  in Salesforce then this is *probably* fine.

File: src/utils.ts [hunk 4]
```diff
@@ -7,125 +16,157 @@ import defaultLeadLayout from "./resources/default_layout/lead.json";
...
-        .with(["Note", "view"], () => {
-            const layout: NoteLayout = settings?.mapping_note
-                ? JSON.parse(settings.mapping_note)
-                : defaultNoteLayout
-                ;
-                
-                return layout.view;
+        .otherwise(() => {
+          return match([curr.errorCode, curr.fields])
+            .with(["REQUIRED_FIELD_MISSING", ["ActivityDateTime"]], () => {
```
Comment (AshleyDawson) [inline src/utils.ts:160]: Hard coded field, `ActivityDateTime`? what happens if someone changes are removes this field in Salesforce? Is that even possible? If it's not possible to remove/change this field  in Salesforce then this is *probably* fine.

**Review by AshleyDawson — CHANGES_REQUESTED** [review]

File: src/pages/create/Activity.tsx (hunk 1 above)
Comment (AshleyDawson) [reply src/pages/create/Activity.tsx]: @iTweeno could you also see if we can lint for use of `console.*()`? Could do with CI catching this sort of thing...

File: src/utils.ts (hunk 4 above)
Comment (amDavid01) [reply src/utils.ts:160]: This is what salesforce returns when you don't insert the date.
No you can't remove this field, it belongs to the default settings 
image

File: src/utils.ts (hunk 3 above)
Comment (amDavid01) [reply src/utils.ts:163]: same answer as below

**Review by AshleyDawson — APPROVED** [review]

**Pull request merged**